src/frontend/components/app/branding-selector.tsx

```diff
@@ -0,0 +1,77 @@
+import {
+  Box,
+  combineStyles,
+  cssClass,
+  selectStyles,
+} from '@adminjs/design-system';
+import omit from 'lodash/omit';
+import React, { FC, useCallback, useState } from 'react';
+import Select from 'react-select';
+import styled, { DefaultTheme, withTheme } from 'styled-components';
+import { AdminJSOptions } from '../../../adminjs-options.interface';
+import { useLocalStorage } from '../../hooks/use-local-storage';
+import { useBranding } from '../../utils/branding-provider';
+
+const SelectContainer = styled(Box)``;
+
+SelectContainer.defaultProps = {
+  className: cssClass('BrandingSelector'),
+};
+
+interface SelectableBranding {
+  value: Required<AdminJSOptions['branding']>;
+  label: string;
+}
+
+interface BrandingSelector {
+  theme: DefaultTheme;
+}
+
+const BrandingSelector: FC<BrandingSelector> = (props) => {
+  const { theme } = props;
+  const styles = selectStyles(theme);
+  const [selected, setSelected] = useState<SelectableBranding>();
+  const [, availableBrandings] = useBranding();
+  const [, storeTheme] = useLocalStorage<DefaultTheme | null>(
+    'adminjs-theme',
+    null
+  );
+  const handleChange = useCallback(async (event: SelectableBranding) => {
+    const branding =
+      typeof event.value === 'function' ? await event.value() : event.value;
+    const combinedTheme = combineStyles(omit(branding.theme, 'details'));
+    storeTheme(combinedTheme as DefaultTheme);
+    setSelected(event);
+    window.location.reload();
+  }, []);
+
+  if (!availableBrandings?.length) return null;
+
+  const selectableBrandings = availableBrandings.reduce((acc, curr) => {
+    if (curr && typeof curr !== 'function') {
+      const selectable: SelectableBranding = {
+        value: curr,
+        label: curr.theme?.details?.name || 'Undefined theme',
+      };
+      acc.push(selectable);
+    }
+    return acc;
+  }, [] as SelectableBranding[]);
+
+  if (!selectableBrandings.length) return null;
+
+  return (
+    <SelectContainer padding="xl">
+      <Select
+        placeholder="Select theme..."
```

Comment:
This should go through i18n imo

Reply:
Moved as message `brandingSelectorSearch`